From 05d7afbe0c2a62b1c67e879e1c14a980d505c9aa Mon Sep 17 00:00:00 2001
From: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
Date: Fri, 26 Jul 2024 02:54:52 +0300
Subject: [PATCH 44/56] dt-bindings: display: bridge: Add schema for Synopsys
 DW HDMI QP TX

Add dt-binding schema containing the common properties for the Synopsys
DesignWare HDMI QP TX controller.

Note this is not a full dt-binding specification, but is meant to be
referenced by platform-specific bindings for this IP core.

Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 .../display/bridge/synopsys,dw-hdmi-qp.yaml   | 42 +++++++++++++++++++
 1 file changed, 42 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/bridge/synopsys,dw-hdmi-qp.yaml

diff --git a/Documentation/devicetree/bindings/display/bridge/synopsys,dw-hdmi-qp.yaml b/Documentation/devicetree/bindings/display/bridge/synopsys,dw-hdmi-qp.yaml
new file mode 100644
index 000000000..f2135f0f2
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/bridge/synopsys,dw-hdmi-qp.yaml
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/bridge/synopsys,dw-hdmi-qp.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Common Properties for Synopsys DesignWare HDMI QP TX Controller IP
+
+maintainers:
+  - Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
+
+description: |
+  This document defines device tree properties for the Synopsys DesignWare
+  HDMI 2.1 Quad-Pixel (QP) TX controller IP core.
+  It doesn't constitute a device tree binding specification by itself, but
+  is meant to be referenced by platform-specific device tree bindings.
+
+  When referenced from platform device tree bindings, the properties defined
+  in this document are defined as follows. The platform device tree bindings
+  are responsible for defining whether each property is required or optional.
+
+properties:
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 6
+    items:
+      - description: Peripheral/APB bus clock
+    additionalItems: true
+
+  clock-names:
+    maxItems: 6
+    items:
+      - const: pclk
+    additionalItems: true
+
+  interrupts:
+    minItems: 1
+    maxItems: 5
+
+additionalProperties: true
-- 
2.39.5

